### PR TITLE
Assets search suggestions

### LINKED ISSUE #232: Search Functionailty (Make dicussions & discord visible)
## (WIP)
Part of #159 - Asset search
Related to #2 - Searchable bevy

Based on being active in the bevy discord help channel, some of the questions asked have already been discussed. In order to intrinsically prevent duplicates, we should use the website as a medium to redirect users for help.

Goal: 
Encourage discovery though:
- [Discussions](https://github.com/bevyengine/bevy/discussions)
- [Discord](https://discord.gg/bevy)
- Cheatbook (https://bevy-cheatbook.github.io)
in order to unify help in one place for new-comers and to promote bevy assets

Pros & Cons:
```diff
+  Not having the costs of hosting a database
-  Not as customizable as it relies on GitHub and crates.io APIs
```

Requires metadata tags from github and or crates.io specified in #230 

[Prototype](https://www.figma.com/file/CZ1svP4l70eMuPYXIOK9d3/bevy_design_sheet?node-id=612%3A58):

Where the URL looks something like `https://bevyengine.org/assets?q=hide+mouse`

And clicking on _Search for “hide mouse” in discussions_ links to: 
[`https://github.com/bevyengine/bevy/discussions?discussions_q=hide+mouse`](https://github.com/bevyengine/bevy/discussions?discussions_q=hide+mouse) and 
[`https://bevy-cheatbook.github.io/?search=hide%20mouse`](https://bevy-cheatbook.github.io/?search=hide%20mouse)
 for bevy-cheatbook

Group 88

### PR DESCRIPTION
**Pull request by BD103**
Closes #232, adopted from #735.

This adds buttons to the bottom of the assets page that let you search in Github Discussions, docs.rs, Bevy Cheatbook, and Discord. See #735 for original PR and a nice video demo.

### Changes from original PR

- Implemented debouncing
- Made buttons always show, even when nothing in the search bar
- Applied Alice's wording for help text
- Fixed icon background on hover and vertical alignment, thanks to doup's suggestion
- Made API docs link to docs.rs, not dev-docs.bevyengine.org

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
@doup could I get your review over here? Also @d-bucur if you're interested :)

**Comment by TrialDragon** [comment]
I can't tell from mobile, but did you fork the original PR so that the original author still gets credit, or I'd this a completely different alternative?

File: templates/macros/assets.html [hunk 1]
```diff
@@ -84,3 +84,11 @@
     {% for s in root.subsections %}{{ self::assets_menu_row(prefix=prefix, section_path=s) }}{% endfor %}
   </ul>
 {% endmacro %}
+{% macro suggestion(id, img, text, url) %}
+<a class="suggestion-item" id="{{ id }}" href="{{ url }}" target="_blank">
+  <div class="suggestion-item__img_wrapper">
+    <img src="{{ img }}" alt="{{ text }} Icon" height="24">
```
Comment (TrialDragon) [inline templates/macros/assets.html:90]:
```suggestion
    <img src="{{ img }}" alt="{{ text }} Icon" width="24" height="24" />
```
You need both width and height for djlint.

**Review by TrialDragon — CHANGES_REQUESTED** [review]

File: templates/macros/assets.html (hunk 1 above)
Comment (BD103) [reply templates/macros/assets.html]: Thanks! I thought the original message only needed one or the other, though that doesn't make sense now that I think about it... 😅

Fixed in e487e13c5d10087386be2875214ea734c8ef1506.

**Review by TrialDragon — APPROVED** [review]
LGTM from the code. (Can't check on local because assets takes hours to generate locally for me for some reason, so can't see the assets page to properly review this).

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**